diff --git a/lib/Core.js b/lib/Core.js
index c0f282ca9e3ddd93e44a62e1d05c1cc45b9f5c6a..77af02a8dc72a0da6c426dc62a0957eb9dacdf57 100644
--- a/lib/Core.js
+++ b/lib/Core.js
@@ -167,7 +167,31 @@ var typeCheckExamples = function (modules) {
     return function_1.pipe(getExampleFiles(modules), RTE.chain(handleImports), RTE.chain(function (examples) {
         return examples.length === 0
             ? cleanExamples
-            : function_1.pipe(writeExamples(examples), RTE.chain(function () { return spawnTsNode; }), RTE.chain(function () { return cleanExamples; }));
+            : function_1.pipe(writeExamples(examples), RTE.chain(function () { return writeFile({"path": "./docs/examples/tsconfig.json", "content": JSON.stringify({
+                "compilerOptions": {
+                  "noEmit": true,
+                  "strict": true,
+                  "noImplicitAny": true,
+                  "noImplicitThis": true,
+                  "strictNullChecks": true,
+                  "strictFunctionTypes": true,
+                  "noImplicitReturns": false,
+                  "noUnusedLocals": false,
+                  "noUnusedParameters": false,
+                  "noFallthroughCasesInSwitch": true,
+                  "allowImportingTsExtensions": true,
+                  "moduleResolution": "node",
+                  "target": "ES2021",
+                  "lib": ["ES2021"],
+                  "paths": {
+                    "@effect/schema": ["../../src/index.ts"],
+                    "@effect/schema/test/*": ["../../test/*"],
+                    "@effect/schema/examples/*": ["../../examples/*"],
+                    "@effect/schema/*": ["../../src/*"]
+                  }
+                }
+              }
+              )}); }), RTE.chain(function () { return spawnTsNode; }), RTE.chain(function () { return cleanExamples; }));
     }));
 };
 // -------------------------------------------------------------------------------------
diff --git a/lib/index.js b/lib/index.js
index 51bdb346c4c33e8835bbda0c16aea71bc0c3115e..7c67a9eec30171b7f4cad959becfab86dd99151c 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -33,7 +33,14 @@ exports.exit = TE.fold(onLeft, function () { return onRight; });
  * @internal
  */
 exports.compilerOptions = {
-    strict: true
+    strict: true,
+    allowImportingTsExtensions: true,
+    paths: {
+      "@effect/schema": ["./src/index.ts"],
+      "@effect/schema/test/*": ["./test/*"],
+      "@effect/schema/examples/*": ["./examples/*"],
+      "@effect/schema/*": ["./src/*"]
+    }
 };
 var capabilities = {
     example: Example_1.Example,